yt/frontends/nc4_cm1/data_structures.py

```diff
@@ -0,0 +1,233 @@
+        # This includes the cosmological parameters.
+        #
+        #   self.unique_identifier      <= unique identifier for the dataset
+        #                                  being read (e.g., UUID or ST_CTIME)
+        self.unique_identifier = int(os.stat(self.parameter_filename)[stat.ST_CTIME])
+        self.parameters = {}  # code-specific items
+        with self._handle.open_ds() as ds:
+            dims = [ds.dimensions[i].size for i in ["xh", "yh", "zh"]]
+            coords = {i: ds.variables[i][:] for i in ["xh", "yh", "zh"]}
+
+            # TO DO: eneralize this to be coordiante variable name agnostic in order to
+            # make useful for WRF or climate data. For now, we're hard coding for CM1
+            # specifically and have named the classes appropriately
+            xh, yh, zh = [ds.variables[i][:] for i in ["xh", "yh", "zh"]]
+            self.domain_left_edge = np.array(
+                [xh.min(), yh.min(), zh.min()], dtype="float64"
+            )
+            self.domain_right_edge = np.array(
+                [xh.max(), yh.max(), zh.max()], dtype="float64"
+            )
+
+            # loop over the variable names in the netCDF file, record only those on the
+            # "zh","yh","xh" grid.
+            varnames = []
+            for key in ds.variables.keys():
+                vardims = ds.variables[key].dimensions
+                if all(x in vardims for x in ["time", "zh", "yh", "xh"]) is True:
+                    varnames.append(key)
+            self.parameters["variable_names"] = varnames
+            self.parameters["lofs_version"] = ds.cm1_lofs_version
+            self.parameters["is_uniform"] = ds.uniform_mesh
+            self.current_time = ds.variables["time"][:][0]
+
+            # record the dimension metadata
+            dim_info = OrderedDict()
+            for dim, meta in ds.dimensions.items():
+                dim_info[dim] = meta.size
+            self.parameters["dimensions"] = dim_info
+
+        self.parameters["coords"] = coords
+
+        self.dimensionality = 3
+        self.domain_dimensions = np.array(dims, dtype="int64")
+        self.periodicity = (False, False, False)
+        self.parameters["time"] = self.current_time
+
+        # Set cosmological information to zero for non-cosmological.
+        self.cosmological_simulation = 0.0
+        self.current_redshift = 0.0
+        self.omega_lambda = 0.0
+        self.omega_matter = 0.0
+        self.hubble_constant = 0.0
+
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
+        # This accepts a filename or a set of arguments and returns True or
+        # False depending on if the file is of the type requested.
+
+        warn_netcdf(args[0])
+        try:
+            from netCDF4 import Dataset
+
+            with Dataset(args[0], "r", keepweakref=True) as ds:
+                is_cm1_lofs = hasattr(ds, "cm1_lofs_version")
+                is_cm1 = hasattr(ds, "cm1 version")
```

Comment:
I'm guessing we're missing an underscore here ?
```suggestion
                is_cm1 = hasattr(ds, "cm1_version")
```

Reply:
weirdly enough, it's not actually a typo... The standard netcdf output has this global attribute `"cm1 version"` with the space.

Reply:
wow I didn't expect that. I think this warrants an inline comment.

Reply:
ya, it's odd. @keltonhalbert , can you confirm this is right? I'm going off of the standard CM1 data file that Leigh shared with us.

Reply:
@chrishavlin it is odd, yes, but unfortunately that is the way the original, untouched-by-us dataset is for the wider modeling community that uses this tool. 
<img width="423" alt="Screen Shot 2020-09-30 at 1 49 50 PM" src="https://user-images.githubusercontent.com/1411265/94727333-318d9380-0324-11eb-8a56-eb39cb679f7b.png">

Reply:
added a short inline comment noting that the space should be there.